### PR TITLE
Update sidebar resizer

### PR DESCRIPTION
**Pull request by uahnbu**
## Motivation

* Sidebar edge too small to grab.
* Setting sidebar doesn't update along with `Ctrl+B`.
* Shrinking icons when resizing.
* Laggy transition when changing label of minimized sidebar.
* Various repeating codes.

## Changes

* Removed sidebar edge element, added edge listener.
* Gathered resizing functions.
* Fixed icon sizes.
* Reduced fast searching time to `400ms`.

<a href="https://gitpod.io/#https://github.com/kimlimjustin/xplorer/pull/219"></a>

### CODE REVIEW

File: src/Components/Layout/resizer.ts [hunk 1]
```diff
@@ -1,78 +1,75 @@
...
-		}
-	});
+const SENSOR = 10;
+const MIN_SIZE = 70;
+const MIN_SNAP = 220;
+
+let sidebar: HTMLElement;
+let settingsSidebar: HTMLElement;
+let xplorerBrand: HTMLElement;
+let appearance: any;
```
Comment (uahnbu) [inline src/Components/Layout/resizer.ts:10]: I would like to assign `type Appearance` like the returned type of `Storage.get('appearance')` as on *line 40* to `appearance` but I don't know where to refer to.

**Review by kimlimjustin — CHANGES_REQUESTED** [review]
```patch
diff --git a/src/Api/storage.ts b/src/Api/storage.ts
index e1f6c33..109c023 100644
--- a/src/Api/storage.ts
+++ b/src/Api/storage.ts
@@ -1,13 +1,10 @@
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { invoke } from '@tauri-apps/api';
-
-interface StorageData {
-	[key: string]: JSON;
-}
+import IStorageData from '../Typings/storageData';
 
 // Store fetched data into variable
-const data: StorageData = {};
+const data: IStorageData = {};
 
 /**
  * Set information to local storage
diff --git a/src/Components/Layout/resizer.ts b/src/Components/Layout/resizer.ts
index 84ed66c..61d01be 100644
--- a/src/Components/Layout/resizer.ts
+++ b/src/Components/Layout/resizer.ts
@@ -1,3 +1,4 @@
+import IStorageData from '../../Typings/storageData';
 import Storage from '../../Api/storage';
 
 const SIDEBAR_EDGE_SENSOR = 10;
@@ -8,7 +9,7 @@ const WINDOW_MIN_SIZE = 800;
 let sidebar: HTMLElement;
 let settingsSidebar: HTMLElement;
 let xplorerBrand: HTMLElement;
-let appearance: any;
+let appearance: IStorageData;
 
 export const resizeSidebar = function (size?: string) {
 	if (!size) {

```

Could you please apply this patch for your comment about the type returned of `Storage.get('appearance')` you mentioned? I tried to push but it gives me permission error. Thanks!

**Comment by uahnbu** [comment]
@kimlimjustin Oh I might have accidentally clicked the disallowing modification option before. Btw I cannot find `storageData` in `../Typings`. Could you provide me with the content of that file?
Hmm. It's weird that my commits after PR also get into this PR. Maybe next time I should create a new branch and push commits to it first?

**Comment by kimlimjustin** [comment]
@uahnbu for the `storageData.d.ts`, it's a new file with following content:

```d.ts
export default interface IStorageData {
	[key: string]: any;
}
```

> I'll have to make a new commit with these changes, right?

Yes, but don't worry bout it, I can squash these commits into one commit while merging it

> Hmm, it's weird that my 3 commits after RP also get into this PR (rest assured they're all checked). Maybe next time I should create a new branch and push commits to it first?

yes, you should create different branches for different tasks, but I think it's okay for this time.

Also, I really appreciate your contributions, thanks a lot!

**Comment by uahnbu** [comment]
Hi @kimlimjustin, you're welcome. I'm quite excited about the project.
Btw, I've made the changes as requested. I still have some questions about PR, would you mind to answer them?
1. Were you able to apply the committed code and debug the app to check whether it works fine, or you must clone all my forked repo, as I can see that you must have separated folder `a/...` and `b/...` to compare right?
2. What is your opinion about multiple commits? For example I made one single error fix about shortcuts. Should I combine it with my changes about sidebar, or should I make it a separated commit?

**Review by kimlimjustin — APPROVED** [review]

**Pull request merged**